### DISCUSSION TITLE
NextJS: Pre-cache configuration

### DISCUSSION THREAD

**Discussion by DanWebb** [serwist/serwist#181] [Q&A]
The default suggested config for NextJS leads to what seems like every file for every page being downloaded when users first land on the site.

In this case, it leads to 3.6MB transferred with an SW, instead of 1.3 MB without.

![Screenshot 2024-08-07 at 11 12 12](https://github.com/user-attachments/assets/2bbce3ae-a5ed-471f-8938-c586e55c3575)

Is this expected? It doesn't seem ideal, especially for people on mobile connections who end up paying for the extra download.

Should this be the recommended setup from the docs with no alternative or suggestion it might lead to larger downloads?

Previously, I disabled the preaching with `disablePrecacheManifest` to stop this from happening, but that option has been removed now.

I'm struggling to find alternative precache options from the documentation, it'd be nice to be able to only pre-cache the app shell.

**Comment by DuCanhGH** [comment]
@DanWebb  set `exclude` to `[() => true]`. This will remove everything from the precache list.

> people on mobile connections who end up paying for the extra download.

The idea is that they would download those files once, and then those files will be reused until you deploy a new update. 3 MB worth of files isn't that much, if you think about it.

To be honest though, optimize your JS like... dawg 💀 Wouldn't catch me shipping a whopping 3 MB total of JS... Unless a 3D library is involved, perhaps.

**Reply by DanWebb** [reply]
I didn't say 3MB of JS was shipped...the precache includes all other asset types.

I'll try exclude out, thanks.

**Reply by DuCanhGH** [reply]
@DanWebb ah, sorry 😅 Well, again, I'd say that 3 MB is not exactly significant, and the benefits of precaching can be great. You do you though.

`exclude` is the way. `disablePrecacheManifest` was some internal stuff meant to short circuit `getFileManifestEntries`'s logic back when I just forked Workbox. Perhaps there's a better way to do this too, but yeah, it's not meant to be used :)
